confidant/scripts/archive.py

```diff
@@ -0,0 +1,169 @@
+        Option(
+            '--days',
+            dest='days',
+            type=int,
+            help=('Permanently archive disabled credentials last modified'
+                  ' greater than this many days (mutually exclusive with'
+                  ' --ids)'),
+        ),
+        Option(
+            '--force',
+            action='store_true',
+            dest='force',
+            default=False,
+            help=('By default, this script runs in dry-run mode, this option'
+                  ' forces the run and makes the changes indicated by the'
+                  ' dry run'),
+        ),
+        Option(
+            '--ids',
+            dest='ids',
+            help=('Archive a comma separated list of credential IDs. (mutually'
+                  ' exclusive with --days)'),
+        ),
+    ]
+
+    def credential_in_service(self, _id, services):
+        for service in services:
+            if _id in service.credentials:
+                return True
+        return False
+
+    def save(self, saves, force=False):
+        _saves = ', '.join([save.id for save in saves])
+        if not force:
+            logger.info(
+                'Would have archived credential and revisions: {}'.format(
+                    _saves,
+                )
+            )
+            return
+        logger.info(
+            'Archiving credential and revisions: {}'.format(
+                _saves,
+            )
+        )
+        with CredentialArchive.batch_write() as batch:
+            for save in saves:
+                batch.save(save)
+
+    def delete(self, deletes, force=False):
+        _deletes = ', '.join([delete.id for delete in deletes])
+        if not force:
+            logger.info(
+                'Would have deleted credential and revisions: {}'.format(
+                    _deletes,
+                )
+            )
+            return
+        logger.info(
+            'Deleting credential and revisions: {}'.format(
+                _deletes,
+            )
+        )
+        with Credential.batch_write() as batch:
+            for delete in deletes:
+                batch.delete(delete)
+
+    def archive(self, credentials, force):
+        services = [
+            service for service in Service.data_type_date_index.query('service')
+        ]
+        for credential in credentials:
+            if self.credential_in_service(credential.id, services):
+                msg = ('Skipping archival of disabled credential {}, as it'
+                       ' is still mapped to a service.')
+                logger.warning(msg.format(credential.id))
+                continue
+            saves = []
+            deletes = []
+            # save the current record.
+            archive_credential = CredentialArchive.from_credential(
+                credential,
+            )
+            saves.append(archive_credential)
+            # fetch and save every revision
+            _range = range(1, credential.revision + 1)
```

Comment:
thoughts on putting the next few 5 lines into a method in credentialmanager? Because I was going to do that myself for https://github.com/lyft/confidant/pull/272

Reply:
Yep. Can do.